File: src/main/java/net/frozenorb/potpvp/util/nametag/listener/NameTagListener.java
package net.frozenorb.potpvp.util.nametag.listener;

import net.frozenorb.potpvp.PotPvPRP;
import net.frozenorb.potpvp.util.packet.ScoreboardTeamPacketMod;
import net.minecraft.server.v1_8_R3.PacketPlayOutScoreboardTeam;
import org.bukkit.Bukkit;
import org.bukkit.craftbukkit.v1_8_R3.entity.CraftPlayer;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.EventPriority;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;
import org.bukkit.event.player.PlayerQuitEvent;
import org.bukkit.metadata.FixedMetadataValue;

import java.lang.reflect.Field;
import java.util.Collections;

public class NameTagListener implements Listener {

    private final PotPvPRP plugin = PotPvPRP.getInstance();

    @EventHandler(priority = EventPriority.HIGH)
    public void onPlayerJoin(PlayerJoinEvent event) {
        event.getPlayer().setMetadata("potpvp-LoggedIn", new FixedMetadataValue(plugin, true));

        plugin.getNameTagHandler().initiatePlayer(event.getPlayer());
        plugin.getNameTagHandler().reloadPlayer(event.getPlayer());
        plugin.getNameTagHandler().reloadOthersFor(event.getPlayer());
    }

    @EventHandler
    public void onPlayerQuit(PlayerQuitEvent event) {
        event.getPlayer().removeMetadata("potpvp-LoggedIn", plugin);
        plugin.getNameTagHandler().getTeamMap().remove(event.getPlayer().getName());
    }

}

File: src/main/kotlin/net/frozenorb/potpvp/kt/nametag/NametagListener.kt
package net.frozenorb.potpvp.kt.nametag

import net.frozenorb.potpvp.PotPvPRP
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerQuitEvent
import org.bukkit.metadata.FixedMetadataValue
import org.bukkit.metadata.MetadataValue
import org.bukkit.event.player.PlayerJoinEvent

internal class NametagListener : Listener {
    @EventHandler
    fun onPlayerJoin(event: PlayerJoinEvent) {
        event.player.setMetadata("potpvp-LoggedIn", FixedMetadataValue(PotPvPRP.getInstance(), true) as MetadataValue)
        PotPvPRP.getInstance().nameTagHandler.initiatePlayer(event.player)
        PotPvPRP.getInstance().nameTagHandler.reloadPlayer(event.player)
        PotPvPRP.getInstance().nameTagHandler.reloadOthersFor(event.player)
    }

    @EventHandler
    fun onPlayerQuit(event: PlayerQuitEvent) {
        event.player.removeMetadata("potpvp-LoggedIn", PotPvPRP.getInstance())
        PotPvPRP.getInstance().nameTagHandler.teamMap.remove(event.player.name)
    }
}